test/src/kbasesearchengine/test/parse/ObjectParserTest.java

```diff
@@ -0,0 +1,256 @@
+					IdMapper.mapKeys(parsingRules.getPrimaryKeyPath(), parsingRules.getRelationRules(), subJts,
+							idConsumer);
+				}
+			}
+			GUID id = ObjectParser.prepareGUID(parsingRules, ref, path, idConsumer);
+			idList.add(id);
+		}
+
+		return idList;
+	}
+
+	@SuppressWarnings({ "static-access", "unchecked" })
+	@Test
+	public void parseSubObjectsGenomeTest() throws Exception {
+
+		String jsonResource = "genome01";
+		String type = "Genome";
+		String guid = "WS:1/1/1";
+
+		Map<GUID, String> guidToJson = this.parseSubObjects(type, jsonResource, guid);
+
+		assertThat(guidToJson.size(), is(1));
+		String jsonString = guidToJson.values().iterator().next();
+		HashMap<String, Object> result = new ObjectMapper().readValue(jsonString, HashMap.class);
+		
+		HashMap<String, Object> expectedResult = new HashMap<String, Object>();
+		expectedResult.put("features", 3);
+		expectedResult.put("domain", "B");
+		expectedResult.put("assembly_ref", "1/2/1");
+		expectedResult.put("scientific_name", "Shewanella");
+		expectedResult.put("id", "NewGenome");
+		
+		assertThat(expectedResult, is(result));
+	}
+	
+	@SuppressWarnings({ "static-access", "unchecked" })
+	@Test
+	public void parseSubObjectsGenomeFeatureTest() throws Exception {
+
+		String jsonResource = "genome01";
+		String type = "GenomeFeature";
+		String guid = "WS:1/1/1";
+
+		Map<GUID, String> guidToJson = this.parseSubObjects(type, jsonResource, guid);
+
+		assertThat(guidToJson.size(), is(3));
+
+		List<String> expectedIds = Arrays.asList("NewGenome.CDS.6211", "NewGenome.CDS.6210", "NewGenome.CDS.6209");
+		for (GUID guidKey : guidToJson.keySet()) {
+			assertThat(expectedIds, hasItem(guidKey.getSubObjectId()));
+			assertThat(guidKey.getSubObjectType(), is("feature"));
+			assertThat(guidKey.getStorageCode(), is("WS"));
+			assertThat(guidKey.getVersion(), is(1));
+			assertThat(guidKey.getAccessGroupId(), is(1));
+			assertThat(guidKey.getAccessGroupObjectId(), is("1"));
+		}
+
+		List<String> expectedKeys = Arrays.asList("ontology_terms", "function", "protein_translation", "location", "id", "type");
+		for (String jsonString : guidToJson.values()) {
+			HashMap<String, Object> result = new ObjectMapper().readValue(jsonString, HashMap.class);
+			assertThat(expectedKeys.containsAll(result.keySet()), is(true));
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	public static Map<GUID, String> parseSubObjects(String type, String jsonResource, String guidString)
+			throws Exception {
+
+		GUID guid = new GUID(guidString);
+
+		InputStream inputStream = ObjectParserTest.class.getResourceAsStream(jsonResource + ".json.properties");
+		Reader reader = new InputStreamReader(inputStream);
+
+		UObject data = UObject.fromJsonString(CharStreams.toString(reader));
+        String name = "TestObj" + "_" + type;
+        String creator = "creator";
+        
+        SourceData obj = SourceData.getBuilder(data, name, creator).withNullableCommitHash(null)
```

Comment:
No need to call `.with` methods with a null value, it's a noop

Reply:
👍